fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/TransactionService.java

```diff
@@ -61,15 +61,46 @@ public ResponseEntity listTransactions(SessionEntity sessionEntity,
             return new ResponseEntity<>(ManualMapper.fromTppToFintech(new TppListTransactionsMock().getTransactionsResponse()), HttpStatus.OK);
         }
+        UUID xRequestId = UUID.fromString(restRequestContext.getRequestId());
 
-        ResponseEntity<TransactionsResponse> transactions = tppAisClient.getTransactions(
+        ResponseEntity<TransactionsResponse> transactions = requestGetTransactions(sessionEntity, bankId, accountId,
+                                                                                   dateFrom, dateTo, entryReferenceFrom,
+                                                                                   bookingStatus, deltaList, redirectCode,
+                                                                                   xRequestId);
+        switch (transactions.getStatusCode()) {
+            case OK:
+                return new ResponseEntity<>(ManualMapper.fromTppToFintech(transactions.getBody()), HttpStatus.OK);
+            case ACCEPTED:
+                log.info("create redirect entity for lot for redirect code {}", redirectCode);
+                redirectHandlerService.registerRedirectStateForSession(redirectCode, fintechOkUrl, fintechNOkUrl);
+                return handleAccepted(sessionEntity, transactions.getHeaders());
+            case UNAUTHORIZED:
+                return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+            default:
+                throw new RuntimeException("DID NOT EXPECT RETURN CODE:" + transactions.getStatusCode());
+        }
+    }
+
+    private ResponseEntity<TransactionsResponse> requestGetTransactions(SessionEntity sessionEntity,
+                                                                        String bankId,
+                                                                        String accountId,
+                                                                        LocalDate dateFrom,
+                                                                        LocalDate dateTo,
+                                                                        String entryReferenceFrom,
+                                                                        String bookingStatus,
+                                                                        Boolean deltaList,
+                                                                        String redirectCode,
+                                                                        UUID xRequestId) {
+        return tppAisClient.getTransactions(
                 accountId,
-                tppProperties.getFintechID(),
                 tppProperties.getServiceSessionPassword(),
                 sessionEntity.getLoginUserName(),
                 RedirectUrlsEntity.buildOkUrl(uiConfig, redirectCode),
                 RedirectUrlsEntity.buildNokUrl(uiConfig, redirectCode),
-                UUID.fromString(restRequestContext.getRequestId()),
+                xRequestId,
+                null,
```

Comment:
Please add a comment about why the values are null, or add some constant like:
`public static final LocalDateTime COMPUTE_X_TIMESTAMP_UTC = null`
and use it here instead of null

Reply:
3 constants were added, with comments regarding the meaning of null values